### ISSUE TITLE
Unit test for decommission error

### ISSUE THREAD

**Issue by alinadonisa** [cockroachdb/cockroach-operator#581]
We should add an unit test that would test that the decommission will throw an error on failure

**Comment by alinadonisa** [comment]
@udnay  I have spoke with @chrislovecnm  and because this task implies a bit of refactory on the decommission logic we will implement this later

**Comment by udnay** [comment]
Makes sense, can you just write some pseudocode for what the test should look like?
